### ISSUE TITLE
`python311Packages.onnxruntime` fails to build, x86_64 24.05

### ISSUE THREAD

**Issue by jarrodsfarrell** [NixOS/nixpkgs#284145]
[/pkgs/development/libraries/onnxruntime/default.nix@master-9c8d5c68](https://github.com/NixOS/nixpkgs/blob/9c8d5c68cc55053749f36ee937ea70c3fe397a80/pkgs/development/libraries/onnxruntime/default.nix)

### Build log

Meaningful errors
```
/build/source/onnxruntime/core/graph/function_utils.cc: In function 'std::unique_ptr<onnx::OpSchema> onnxruntime::function_utils::CreateSchema(const onnxruntime::Graph&, const onnxruntime::IndexedSubGraph&, bool)':
/build/source/onnxruntime/core/graph/function_utils.cc:50:77: error: 'all_tensor_types_with_bfloat' is not a member of 'onnx::OpSchema'
   50 |     op_schema->TypeConstraint("TAggregatedTypes", ONNX_NAMESPACE::OpSchema::all_tensor_types_with_bfloat(),
      |                                                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/onnxruntime/core/graph/function_utils.cc: In function 'void onnxruntime::function_utils::IOTypeConstraintHelper(const onnx::FunctionProto&, std::unique_ptr<onnx::OpSchema>&, const onnxruntime::InlinedHashMap<std::__cxx11::basic_string<char>, int>&, const onnxruntime::InlinedHashMap<std::__cxx11::basic_string<char>, int>&)':
/build/source/onnxruntime/core/graph/function_utils.cc:102:47: error: 'all_tensor_types_with_bfloat' is not a member of 'onnx::OpSchema'
  102 |   all_types.reserve(ONNX_NAMESPACE::OpSchema::all_tensor_types_with_bfloat().size() +
      |                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/onnxruntime/core/graph/function_utils.cc:104:46: error: 'all_tensor_types_with_bfloat' is not a member of 'onnx::OpSchema'
  104 |   all_types.insert(ONNX_NAMESPACE::OpSchema::all_tensor_types_with_bfloat().cbegin(),
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/onnxruntime/core/graph/function_utils.cc:105:46: error: 'all_tensor_types_with_bfloat' is not a member of 'onnx::OpSchema'
  105 |                    ONNX_NAMESPACE::OpSchema::all_tensor_types_with_bfloat().cend());
      |
```
Full logs: [build.log](https://github.com/NixOS/nixpkgs/files/14070388/build.log)

### Notify maintainers

@jonringer @puffnfresh @ck3d @cbourjau

### Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.74, NixOS, 24.05 (Uakari), 24.05pre574351.612f97239e2c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos"`
 - channels(jarrod): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### Additional

Source file with issue: [`onnxruntime/core/graph/function_utils.cc@v1.15.1`](https://github.com/microsoft/onnxruntime/blob/v1.15.1/onnxruntime/core/graph/function_utils.cc)

I tried to locate where `OpSchema` is defined, but was unable to.

**Comment by jarrodsfarrell** [comment]
Edited the issue and I'm not sure if GitHub retroactively sends notifications, so pardon the double-mention: @jonringer @puffnfresh @ck3d @cbourjau

**Comment by mweinelt** [comment]
Will be fixed by #258392.
